sonar/modules/users/cli.py

```diff
@@ -67,9 +67,15 @@ def import_users(infile):
             password = hash_password(password)
             del user_data['password']
 
+            if not user_data.get('roles'):
+                user_data['roles'] = [UserRecord.ROLE_USER]
+
             roles = user_data.get('roles', []).copy()
-            if not roles or not isinstance(roles, list):
-                roles = []
+
+            for role in roles:
+                if not datastore.find_role(role):
```

Comment:
If this take while it can be afterward with a unique list of collected roles.

Reply:
Not really sure, because we need the roles to create user account just after. But maybe yes, we can collect all roles before and create them before creating users.